### PR TITLE
[release-0.36] virt-launcher: [masquerade] Stop filtering dhcp reuqests by vm MAC

### PR DESCRIPTION
**Pull request by AlonaKaplan**
**What this PR does / why we need it**:
Manual cherry pick of https://github.com/kubevirt/kubevirt/pull/4804

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/vmi_networking_test.go [hunk 1]
```diff
@@ -789,7 +789,21 @@ var _ = Describe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][le
...
+				}, 10)).To(Succeed(), "failed to restart the vmi")
+				tests.WaitUntilVMIReady(vmi, loginMethod)
+				if ipFamily == k8sv1.IPv6Protocol {
+					Expect(configureIpv6(vmi)).To(Succeed(), "failed to configure ipv6 on vmi after restart")
+				}
+				Expect(ping(podIP)).To(Succeed())
+			},
+				table.Entry("IPv4", k8sv1.IPv4Protocol),
+				table.Entry("IPv6", k8sv1.IPv6Protocol),
+			)
```
Comment (EdDev) [inline tests/vmi_networking_test.go:806]: We did not have this at that time. The CI will probably fail on this.

Comment (AlonaKaplan) [reply tests/vmi_networking_test.go:806]: Didn't have what?

Comment (EdDev) [reply tests/vmi_networking_test.go:806]: The table.

**Review by EdDev — APPROVED** [review]

**Review by enp0s3 — APPROVED** [review]
/approve

**Comment by rmohr** [comment]
Well known issue where a fix will land soon. Force merge, the PR is fine.

**Pull request merged**